### PR TITLE
fix: error handle property access and instance mutation

### PR DESCRIPTION
**Pull request by saikumarrs**
## PR Description

Enhanced error handling utilities to preserve error types and added defensive programming patterns for better robustness.

### `getMutatedError` Function Improvements
- **Preserve Error Types**: `TypeError` stays `TypeError`, `ReferenceError` stays `ReferenceError`, etc.
- **Immutability**: Creates new error objects instead of mutating originals
- **Property Preservation**: Maintains stack traces and custom error properties
- **Graceful Fallbacks**: Handles constructor failures and edge cases

### `isStorageQuotaExceeded` Defensive Programming
- **Error Resilience**: Added try-catch to prevent crashes from malformed error objects
- **Edge Case Handling**: Safely handles null/undefined inputs and property access failures

## Linear task (optional)

https://linear.app/rudderstack/issue/SDK-3170/spike-error-handler-property-access-issue

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [ ] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling by preserving error types, stack traces, and custom properties when mutating errors.
  * Enhanced storage quota detection to more accurately identify quota exceeded errors across different browsers.

* **Tests**
  * Expanded test coverage for error mutation utilities, ensuring correct type preservation, property copying, and resilience to edge cases.
  * Added comprehensive tests for storage capability detection, including support for various browser-specific scenarios and robustness against invalid inputs.

### CODE REVIEW

File: packages/analytics-js-common/src/utilities/errors.ts [hunk 1]
```diff
@@ -4,9 +4,7 @@ import { stringifyWithoutCircular } from './json';
 const MANUAL_ERROR_IDENTIFIER = '[SDK DISPATCHED ERROR]';
 
 const getStacktrace = (err: any): string | undefined => {
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js-common/src/utilities/errors.ts:6]: It's ok to not check for these.

File: packages/analytics-js/src/components/capabilitiesManager/detection/storage.ts [hunk 2]
```diff
@@ -15,12 +15,16 @@ import {
...
-const isStorageQuotaExceeded = (e: DOMException | any): boolean => {
+const isStorageQuotaExceeded = (e: any): boolean => {
   const matchingNames = ['QuotaExceededError', 'NS_ERROR_DOM_QUOTA_REACHED']; // [everything except Firefox, Firefox]
   const matchingCodes = [22, 1014]; // [everything except Firefox, Firefox]
-  const isQuotaExceededError = matchingNames.includes(e.name) || matchingCodes.includes(e.code);
+  try {
+    const isQuotaExceededError = matchingNames.includes(e.name) || matchingCodes.includes(e.code);
 
-  return e instanceof DOMException && isQuotaExceededError;
+    return e instanceof DOMException && isQuotaExceededError;
```
[the remark this answers is not in the document]
Comment (saikumarrs) [reply packages/analytics-js/src/components/capabilitiesManager/detection/storage.ts]: Addressed.

**Review by MoumitaM — APPROVED** [review]
